### PR TITLE
Adding first version of readme.md

### PR DESCRIPTION
**Pull request by ponomvrenko**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+2. **Order Management:** Once you've filled your shopping cart with the books you desire, our application provides a streamlined order management system. You can review your orders and add them.
+
+3. **Book Management with Pagination and Sorting:** We understand the importance of efficient book discovery. Our application offers a sophisticated search functionality that allows users to find all books by categories, author, title, or by id. With pagination support, you can explore a vast catalog of books without overwhelming your search results.
+
+4. **Security and JWT Tokens for User Auth:** Security is paramount in our application. We've implemented Spring Security to safeguard your data and transactions. JWT (JSON Web Tokens) are used for secure authentication and authorization, ensuring that only authorized users can access sensitive functionalities.
+
+## Technologies Used
+
+Our application leverages several cutting-edge technologies, including:
+
```
Comment (d1sam) [inline readme.md:33]: what about testing technologies? please also write categories, that related to technologies.

File: readme.md
Comment (d1sam) [inline readme.md:1]: Where is example of your endpoints?

File: readme.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+**Issue:** Ensuring data security and user authentication was paramount.
+
+**Solution:** Spring Security and JWT tokens were implemented for robust protection.
+
+### Challenge 3: Exception Handling
+
+**Issue:** Handling errors and exceptions systematically was crucial.
+
+**Solution:** We developed a global handler and custom exceptions for better error reporting.
```
Comment (d1sam) [inline readme.md]: please describe detailer problems and solutions.

File: readme.md
Comment (d1sam) [inline readme.md:1]: please also add Possible improvements section.

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but there are some improvements should be done!

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+**MySQL:** Enables connectivity between the Spring application and MySQL database.
+
+
+***📌Testing:***
+
+**JUnit:** A widely used testing framework for writing and running unit tests.
+
+**HSQLDB (for testing):** An in-memory database used for testing data access logic.
+
+**Spring Boot Starter Test:** Provides testing support for Spring Boot applications.
```
Comment (yevheniisynytsia) [inline readme.md:59]: + Mockito

File: readme.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+- `GET: /api/books` - The endpoint for review all the books with User authority.
+- `GET: /api/books/{id}` - The endpoint for searching a specific book with user authority.
+- `POST: /api/books` - The endpoint for creating a new book with Admin authority.
+- `PUT: /api/books/{id}` - The endpoint for updating information about our book with Admin authority
+- `DELETE: /api/books/{id}` - The endpoint for deleting books with Admin authority.
+- `GET: /api/books/search` - The endpoint for searching books by parameters with User authority.
+
+**CategoryController:** receive and handle requests for adding, updating, getting categories and getting all books by category.
++ `GET: /api/categories` - The endpoint to look at all categories (User authority)
++ `GET: /api/categories/{id}` - The endpoint for searching a specific category (User authority)
```
Comment (yevheniisynytsia) [inline readme.md]: for consistency let's write with User authority

File: readme.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+- Java 17 [Download Java 17](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html)
+- Docker [Download Docker](https://www.docker.com/)
+
+## ⬇️Installation
+
+1. Clone repository:
+   - Open IDEA → File → New Project from Version Control and insert link: [https://github.com/ponomvrenko/online-book-store](https://github.com/ponomvrenko/online-book-store).
+   - Or clone from the console with the command: `git clone https://github.com/ponomvrenko/online-book-store`
+2. Build project and download dependencies for Maven with command: `mvn clean install`
```
Comment (yevheniisynytsia) [inline readme.md:148]: make sure you have docker up and running

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Well done! Small changes left

File: readme.md (hunk 5 above)
Comment (ponomvrenko) [reply readme.md:148]: image

**Review by d1sam — APPROVED** [review]
well done!

**Pull request merged**